packages/auth/src/core/user/user_impl.ts

```diff
@@ -200,6 +201,11 @@ export class UserImpl implements UserInternal {
   }
 
   async delete(): Promise<void> {
+    if (_isFirebaseServerApp(this.auth.app)) {
+      return Promise.reject(
+        _createError(this.auth, AuthErrorCode.OPERATION_NOT_SUPPORTED)
```

Comment:
I'm sorry to not have mentioned this in the earlier atomic PR but is this error code enough information for these, or is it worth using `_errorWithCustomMessage` to add more specific info that it's about FirebaseServerApp? I guess it makes sense if you are trying to minimize the increase in size to the auth bundle. Not a big deal, the error message should point to the line of code and developers can see that it's a FirebaseServerApp issue.

Reply:
Done. The error is now more descriptive, telling users that attempting to change the current user isn't supported with FirebaseServerApp instances.